server/preferences/preference.model.js

```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const PreferencesSchema = new Schema({
+  name: String,
+  value: String,
+  permission: String,
```

Comment:
Is permission going to be a name-string of a class of permissions or a role or something? I was thinking a Role might have many permissions, and that a user would therefore have a lot of them.

Reply:
I was thinking a permission would be associated with a specific level of role, such as `Admin`, `Owner`, `Member`, etc. A user might have several roles. A user whose roles are `['Member', Admin]`, would be able to set any preference that has a permission level of `Admin` and under. In this scenario, a user would have an array of roles, but a single preference would have only be set to the lowest permission level. Thoughts?